test/nyc-bin.js

```diff
@@ -101,6 +101,48 @@ describe('the nyc cli', function () {
         done()
       })
     })
+
+    it('should allow negated exclude patterns', function (done) {
+      const args = [bin, '--all', '--exclude', '**/include-exclude/**', '--exclude', '!**/exclude-negated.js', process.execPath, './half-covered.js']
+
+      const proc = spawn(process.execPath, args, {
+        cwd: fixturesCLI,
+        env: env
+      })
+
+      let stdout = ''
+      proc.stdout.on('data', chunk => {
+        stdout += chunk
+      })
+
+      proc.on('close', code => {
+        code.should.equal(0)
+        stdout.should.not.match(/excluded\.js/)
+        stdout.should.match(/exclude-negated\.js/)
+        done()
+      })
+    })
+
+    it('should include \'node_modules\' using exclude patterns', function (done) {
```

Comment:
Lets add another test (in a follow-up) to do nearly the same thing as this, just without the `--all` argument and with a test script that actually has `require('./include-exclude/node_modules/cover-me.js');`?  This will prove that `--exclude '!**/node_modules/**'` works again.  I've already done this basic test at the command-line and it did what I expected.

Reply:
Cool, I'll put something together after lunch